File: backend/routers/timeline.py
"""Timeline routes: screenshots grouped by date."""

from fastapi import APIRouter, Depends

from backend.auth import require_auth
from backend.services.timeline_service import get_timeline, get_timeline_stats

router = APIRouter(prefix="/api/timeline", tags=["timeline"], dependencies=[Depends(require_auth)])


@router.get("")
async def timeline(
    game_id: int | None = None,
    start_date: str | None = None,
    end_date: str | None = None,
    page: int = 1,
    limit: int = 30,
):
    """Get screenshots grouped by date, paginated by days."""
    return await get_timeline(
        game_id=game_id,
        start_date=start_date,
        end_date=end_date,
        page=page,
        limit=limit,
    )


@router.get("/stats")
async def timeline_stats():
    """Get timeline summary statistics."""
    return await get_timeline_stats()

File: frontend/src/api/timeline.ts
import { api } from './client';

export interface TimelineScreenshot {
  id: number;
  filename: string;
  file_path: string;
  thumbnail_path_sm: string | null;
  thumbnail_path_md: string | null;
  taken_at: string | null;
  uploaded_at: string;
  is_favorite: boolean;
  width: number | null;
  height: number | null;
  game_id: number;
  game_name: string;
  has_annotation: boolean;
}

export interface TimelineDay {
  date: string;
  screenshot_count: number;
  games: string[];
  screenshots: TimelineScreenshot[];
}

export interface TimelineResponse {
  days: TimelineDay[];
  total_days: number;
  page: number;
  has_more: boolean;
}

export interface TimelineStats {
  total_screenshots: number;
  total_days: number;
  earliest_date: string | null;
  latest_date: string | null;
  total_games: number;
}

export async function getTimeline(params?: {
  start_date?: string;
  end_date?: string;
  game_id?: number;
  page?: number;
  limit?: number;
}): Promise<TimelineResponse> {
  const searchParams = new URLSearchParams();
  if (params?.start_date) searchParams.set('start_date', params.start_date);
  if (params?.end_date) searchParams.set('end_date', params.end_date);
  if (params?.game_id) searchParams.set('game_id', String(params.game_id));
  if (params?.page) searchParams.set('page', String(params.page));
  if (params?.limit) searchParams.set('limit', String(params.limit));
  const qs = searchParams.toString();
  return api.get(`/timeline${qs ? `?${qs}` : ''}`);
}

export async function getTimelineStats(): Promise<TimelineStats> {
  return api.get('/timeline/stats');
}
